### ISSUE TITLE
EnvelopeRecipientTabs prefillTabs is misleading

### ISSUE THREAD

**Issue by nathany-copia** [neilberkman/docusign_elixir#36]
> "pre-fill tabs cannot be for a specific singer" - @InbarGazit

via https://stackoverflow.com/questions/66963685/is-it-possible-to-prefill-values-for-the-tabs-from-prefilltabs-group-in-docusign

It seems misleading to have prefillTabs on an Envelope Recipient if they can't be used for a specific recipient.

If this is all generated via Swagger, it seems like we need to take it up with DocuSign? But since I'm new to the DocuSign API (and having a terrible time with the documentation), I was hoping for a second opinion first. Also wondering if there have been any updates to the Swagger?

**Comment by nathany-copia** [comment]
I've requested a better example from DocuSign here:
https://stackoverflow.com/questions/75750295/is-there-a-working-example-of-pre-filling-an-existing-docusign-template-with-pre

**Comment by neilberkman** [comment]
Hi @nathany-copia thanks for raising this issue. Since we last synced our code to the 2.1 eSignature OpenAPI spec in 49223163bc3d27a4389f7442b146250cbcacbbda there's been an additional commit in Docusign's repo. I just searched the diff for `profile` and found [this](https://github.com/docusign/OpenAPI-Specifications/commit/f3fc1b7b0b3218cc83e77c09a1182975bbdb6b4a#diff-d083a41811e1eabb331beafc33be876edd2661487904e61f23210f9a943818e5L3107-R3107). Not sure if it would address your issue?

FYI while I maintain this package I'm no longer working on a project that uses it so I generally rely on contributors for anything significant. The README has instructions for updating based on the latest OpenAPI spec although unfortunately, as of the last update, some manual modifications to the Swagger file were necessary. You can see the discussion in #33 and the [last few commits here](https://github.com/neilberkman/docusign_elixir/pull/33/commits). If you are interested in contributing, that may be enough of a guide.

**Comment by nathany-copia** [comment]
@neilberkman Thanks for your help. To be honest, I'm having difficulty parsing the DocuSign repo.

I'd be happy to contribute a Swagger update, but probably not today as I think I need a break from DocuSign (and I also want to wait-and-see if they provide direction on StackOverflow for my specific issue)

**Comment by neilberkman** [comment]
@nathany-copia sounds good, I hope they respond and help to clear things up.

**Comment by InbarGazit** [comment]
@neilberkman I can help you out, was about to respond to SO thread
The name "prefilled tabs" is  unfortunate, and is confusing. They should be named "sender tabs"
You can pre-fill any tab, not just prefill tabs, and you can do this easily using the API.
I will see what you specific issue with title tab is, but just so I'm clear - you can ger regular tab and fill their values, is that correct?

**Comment by neilberkman** [comment]
@InbarGazit I appreciate your help! I'm the maintainer of this package but @nathany-copia opened the issue. He should be able to respond to your questions.

**Comment by nathany-copia** [comment]
@InbarGazit Thank you for your help. Much appreciated.

To provide some context, I was given a template document in DocuSign and asked to figure out how to both email the template (easy enough), and how to prefill the fields with any information we have on hand. I made some progress yesterday, and I'm slowly starting to understand how the DocuSign APIs work, but I'm still missing some things despite reading quite a few DocuSign docs and utilizing the models generated from Swagger in this this library. Maybe there are some API docs that I just haven't discovered yet, and maybe this library's documentation/Swagger isn't up-to-date 🤷🏻‍♂️ .

> The pre-fill tabs are for the sender, if you want recipient tabs, you can set their value just like you did with the text tab.

The document has two roles with Name, Email, Title for each recipient. There are also fields for Term, Price, Date. They were configured as fields of one of the recipients, but it seems more appropriate to me to use prefill fields provided by the sender. That way those fields (Term, Price, Date) could be filled in before the document is sent to both parties. I have access to the DocuSign template to modify the tabs to whatever is most suitable.

I started with the [EnvelopeDefinition](https://hexdocs.pm/docusign/DocuSign.Model.EnvelopeDefinition.html). Right now I'm providing the tempalteId, emailSubject, the [TemplateRole](https://hexdocs.pm/docusign/DocuSign.Model.TemplateRole.html#t:t/0) and a status of `sent`. 

Late yesterday I read your previous comment on StackOverflow: "pre-fill tabs cannot be for a specific singer"

What I haven't been able to figure out yet is how to get the sender to pre-fill tabs. The `EnvelopeDefinition` doesn't have any tabs / prefillTabs that I can see.

Adding to the confusion, as mentioned above, the TemplateRole has [EnvelopeRecipientTabs](https://hexdocs.pm/docusign/DocuSign.Model.EnvelopeRecipientTabs.html#t:t/0) which has [PrefillTabs](https://hexdocs.pm/docusign/DocuSign.Model.PrefillTabs.html#t:t/0), seemingly implying that PrefillTabs are sent with the recipient fields when they are not.

From that old [StackOverflow](https://stackoverflow.com/questions/66963685/is-it-possible-to-prefill-values-for-the-tabs-from-prefilltabs-group-in-docusign) comment, it sounds like I may need to `create` the `EnvelopeDefinition` first without sending it, and then do "something" to prefill the fields before sending it? That's what I intend to look into and experiment with today -- unless you point me in a different direction.

> However, some tabs such as Date, Initial, SignHere, Title, and a few others cannot be set via the API.

The same confusion exists here as well, where [EnvelopeRecipientTabs](https://hexdocs.pm/docusign/DocuSign.Model.EnvelopeRecipientTabs.html#t:t/0) has titleTabs in the API, but it doesn't work.

This is extremely frustrating to work with, because a developer's assumption is that they are doing something wrong, but it's actually the API specification that is wrong, at least for this use case / scenario. (Maybe they are there for creating documents on the fly? Though I don't know if the Value field of titleTabs ever works. 🤷🏻 )

Please clarify, is it not possible to set a Title at all? You said some tabs cannot be set via the API. Name and Email are automatically set based on the [TemplateRole](https://hexdocs.pm/docusign/DocuSign.Model.TemplateRole.html#t:t/0), without specifying them as "tabs" -- but I could not find a Title in the TemplateRole.

**Comment by neilberkman** [comment]
>  maybe this library's documentation/Swagger isn't up-to-date 🤷🏻‍♂️ .

@nathany-copia just making sure you saw this, from my [previous comment](https://github.com/neilberkman/docusign_elixir/issues/36#issuecomment-1470887388):

> Since we last synced our code to the 2.1 eSignature OpenAPI spec in 49223163bc3d27a4389f7442b146250cbcacbbda there's been an additional commit in Docusign's repo. I just searched the diff for `profile` and found [this](https://github.com/docusign/OpenAPI-Specifications/commit/f3fc1b7b0b3218cc83e77c09a1182975bbdb6b4a#diff-d083a41811e1eabb331beafc33be876edd2661487904e61f23210f9a943818e5L3107-R3107). Not sure if it would address your issue?

**Comment by nathany-copia** [comment]
@neilberkman Yes, working on it right now. 🙃 

#37

**Comment by nathany-copia** [comment]
@neilberkman I don't think that update will help with what I'm working on right now, but it's probably good to do anyway.

**Comment by neilberkman** [comment]
OK thanks @nathany-copia I really appreciate it. Are you still testing it?

**Comment by nathany-copia** [comment]
Yes, but I am using a fairly small portion of the API client. Might be good if some other people try it out too.

The diff doesn't look that huge though, so that's good.

**Comment by neilberkman** [comment]
@nathany-copia I merged your API updates and released a while ago. Thanks again for your contributions! Just wondering if you're still searching for a solution for this issue?

**Comment by nathany-copia** [comment]
Thanks @neilberkman.

I'm not actively looking at it. Let's just close the issue.
